thirdeye-server/src/main/java/ai/startree/thirdeye/resources/CrudResource.java

```diff
@@ -318,4 +319,26 @@ public Response deleteAll(@ApiParam(hidden = true) @Auth ThirdEyePrincipal princ
     dtoManager.findAll().forEach(this::deleteDto);
     return Response.ok().build();
   }
+
+  @GET
+  @Path("/count")
+  @Timed
+  public Response countWithPredicate(
+      @ApiParam(hidden = true) @Auth ThirdEyePrincipal principal,
+      @Context UriInfo uriInfo
+  ) {
+    final MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
+    final CountApi api = new CountApi();
+    try {
+      final Long count = queryParameters.size() > 0
+          ? dtoManager.count(new DaoFilterBuilder(apiToIndexMap).buildFilter(queryParameters)
+          .getPredicate())
+          : dtoManager.count();
+      api.setCount(count);
+    } catch (Exception e) {
```

Comment:
Please refer to existing ways of doing this. we shouldn't be responding with 200 in case of an exception

Reply:
My bad. Fixed it.